### PR TITLE
chore: add tests for Scalar, Vector and Direction

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: include/OpenSpaceToolkit/Physics/Data/Vector.hpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ class Vector
 
     Vector inFrame(const Shared<const Frame>& aFrameSPtr, const Instant& anInstant) const;
 
-    String toString(const Integer& aPrecision = Integer::Undefined()) const;
+    String toString(const Integer& aPrecision = 6) const;
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/Physics/Data/Vector.hpp:56]: The default print would raise an error here as Integer::Undefined does not play well with Vector3d printing

File: include/OpenSpaceToolkit/Physics/Data/Direction.hpp [hunk 2]
```diff
@@ -34,10 +34,6 @@ class Direction : public Vector
    public:
     Direction(const Vector3d& aValue, const Shared<const Frame>& aFrameSPtr);
 
-    bool operator==(const Direction& aDirection) const;
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Physics/Data/Direction.hpp:37]: Oh woops I added these back in the other MR, feel free to discard that back

File: src/OpenSpaceToolkit/Physics/Data/Direction.cpp [hunk 3]
```diff
@@ -27,21 +27,6 @@ Direction::Direction(const Vector3d& aValue, const Shared<const Frame>& aFrameSP
...
 }
 
-bool Direction::operator==(const Direction& aDirection) const
-{
-    if ((!this->isDefined()) || (!aDirection.isDefined()))
-    {
-        return false;
-    }
-
-    return Vector::operator==(aDirection);
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Physics/Data/Direction.cpp:37]: What is this line doing? Seems strange, so I guess might as well delete it

File: test/OpenSpaceToolkit/Physics/Data/Scalar.test.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+{
+    {
+        EXPECT_EQ(scalar_.getUnit(), unit_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Physics_Data_Scalar, InUnit)
+{
+    {
+        EXPECT_TRUE(scalar_.inUnit(Unit::Length(Length::Unit::Foot)).isDefined());
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Physics/Data/Scalar.test.cpp:74]: Can you actually check the value here? If you don't want to do the math just 
```suggestion
        EXPECT_TRUE(scalar_.inUnit(Unit::Length(Length::Unit::Foot)).isDefined());
        
        EXPECT_DOUBLE_EQ(value_, scalar_.inUnit(Unit::Length(Length::Unit::Meter)));
```

File: test/OpenSpaceToolkit/Physics/Data/Vector.test.cpp [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+{
+    {
+        EXPECT_EQ(vector_.getFrame(), frameSPtr_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Physics_Data_Vector, InUnit)
+{
+    {
+        EXPECT_TRUE(vector_.inUnit(Unit::Length(Length::Unit::Foot)).isDefined());
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Physics/Data/Vector.test.cpp:91]: ditto
```suggestion
        EXPECT_TRUE(vector_.inUnit(Unit::Length(Length::Unit::Foot)).isDefined());
        
        EXPECT_DOUBLE_EQ(value_.norm(), vector_.inUnit(Unit::Length(Length::Unit::Meter)).norm());
```

File: test/OpenSpaceToolkit/Physics/Data/Vector.test.cpp [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Physics_Data_Vector, InFrame)
+{
+    {
+        // TBI: Can be improved
+        EXPECT_NO_THROW(vector_.inFrame(Frame::ITRF(), Instant::J2000()));
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Physics/Data/Vector.test.cpp:99]: It sure can ;)
```suggestion        
        EXPECT_EQ(vector_.inFrame(frameSPtr_, Instant::J2000(), vector_);
        EXPECT_NE(vector_.inFrame(frameSPtr_, Instant::J2000(), vector_.inFrame(Frame::ITRF(), Instant::J2000()));
```

**Review by apaletta3 — APPROVED** [review]
LGTM. Thanks for taking care of this! Just approve nits if you agree :)

**Pull request merged**